interfaces/builtin/mir.go

```diff
@@ -0,0 +1,606 @@
+rt_sigaction
+rt_sigpending
+rt_sigprocmask
+rt_sigqueueinfo
+rt_sigreturn
+rt_sigsuspend
+rt_sigtimedwait
+rt_tgsigqueueinfo
+sched_getaffinity
+sched_getattr
+sched_getparam
+sched_get_priority_max
+sched_get_priority_min
+sched_getscheduler
+sched_rr_get_interval
+# LP: #1446748 - when support syscall arg filtering, enforce pid_t is 0 so the
+# app may only change its own scheduler
+sched_setscheduler
+
+sched_yield
+
+select
+_newselect
+pselect
+pselect6
+
+semctl
+semget
+semop
+semtimedop
+sendfile
+sendfile64
+sendmsg
+sendto
+
+# snappy doesn't currently support per-app UID/GIDs so don't allow this family
+# of syscalls. To properly support these, we need to have syscall arg filtering
+# (LP: #1446748) and per-app UID/GIDs.
+#setgid
+#setgid32
+#setgroups
+#setgroups32
+#setregid
+#setregid32
+#setresgid
+#setresgid32
+#setresuid
+#setresuid32
+#setreuid
+#setreuid32
+#setuid
+#setuid32
+
+# These break isolation but are common and can't be mediated at the seccomp
+# level with arg filtering
+setpgid
+setpgrp
+
+set_thread_area
+setitimer
+
+# apps don't have CAP_SYS_RESOURCE so these can't be abused to raise the hard
+# limits
+setrlimit
+prlimit64
+
+set_mempolicy
+set_robust_list
+setsid
+set_tid_address
+setsockopt
+setxattr
+fsetxattr
+lsetxattr
+
+shmat
+shmctl
+shmdt
+shmget
+signal
+sigaction
+signalfd
+signalfd4
+sigaltstack
+sigpending
+sigprocmask
+sigreturn
+sigsuspend
+sigtimedwait
+sigwaitinfo
+socket
+socketpair
+splice
+
+stat
+stat64
+fstat
+fstat64
+fstatat64
+lstat
+newfstatat
+oldfstat
+oldlstat
+oldstat
+
+statfs
+statfs64
+fstatfs
+fstatfs64
+statvfs
+fstatvfs
+ustat
+
+symlink
+symlinkat
+
+sync
+sync_file_range
+sync_file_range2
+arm_sync_file_range
+fdatasync
+fsync
+syncfs
+sysinfo
+syslog
+tee
+tgkill
+time
+timer_create
+timer_delete
+timer_getoverrun
+timer_gettime
+timer_settime
+timerfd_create
+timerfd_gettime
+timerfd_settime
+times
+tkill
+
+truncate
+truncate64
+ftruncate
+ftruncate64
+
+umask
+
+uname
+olduname
+oldolduname
+
+unlink
+unlinkat
+
+utime
+utimensat
+utimes
+futimesat
+
+vfork
+vmsplice
+wait4
+oldwait4
+waitpid
+waitid
+
+write
+writev
+pwrite
+pwrite64
+pwritev
+`)
+
+var mirConnectedPlugAppArmor = []byte(`
+# Description: Allow use of the Mir server. Reserved because this
+# gives privileged access to the system.
+# Usage: reserved
+
+
+capability dac_override,
+capability sys_tty_config,
+capability sys_admin,
+
+unix (receive, send) type=seqpacket addr=none,
+/tmp/mir_socket rw,
+/dev/dri/card0 rw,
+/dev/shm/\#* rw,
+
+/sys/devices/**/uevent rw,
+/sys/devices/**/ r,
+/dev/input/* rw,
+/dev/tty* wr,
+/run/udev/data/* r,
+/run/udev/** rw,
+
+ptrace peer=**,
+/bin/sleep mrix,
+/bin/pidof mrix,
+/bin/sed mrix,
+/bin/cp mrix,
+/sbin/killall5 ixr,
+/usr/bin/expr ixr,
+/usr/bin/chmod ixr,
+/bin/chmod ixr,
+/proc/ r,
+/proc/*/stat r,
+/proc/*/cmdline r,
+/sys/bus/ r,
+/sys/class/ r,
+/sys/class/input/ r,
+/sys/class/drm/ r,
+/etc/udev/udev.conf r,
+capability sys_ptrace,
+capability chown,
```

Comment:
Again, this is the server, but I can't recall why the server needs to chown either. Can you comment in the code with a 'Note:'.